src/WebJobs.Script.WebHost/Diagnostics/ISystemEventGenerator.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    public interface ISystemEventGenerator
+    {
+        void LogEvent(TraceLevel level, string subscriptionId, string appName, string functionName, string eventName, string source, string details, string summary, Exception exception = null);
```

Comment:
Yeah I agree in general, but these methods are really only called from two places, so it's not a real usability issue. Not sure it's worth the additional effort to define a new type, and pack/unpack that type at this point.